sapicore/data/external/drift.py

```diff
@@ -0,0 +1,110 @@
+"""UCSD Drift dataset."""
+import os
+import shutil
+
+from zipfile import ZipFile
+
+import torch
+import pandas as pd
+
+from sklearn.datasets import load_svmlight_file
+from scipy.sparse import csr_matrix as csr
+
+from sapicore.data import Data, AxisDescriptor
+
+
+class DriftDataset(Data):
+    """Processing logic for the UCSD drift dataset.
+
+    Implements the protected :meth:`~data.Data._standardize` to extract, transform, and save samples and labels
+    from the raw .dat files downloaded from the UCI repository.
+
+    See Also
+    --------
+    `Drift dataset specification <https://archive.ics.uci.edu/ml/datasets/Gas%20Sensor%20Array%20Drift%20Dataset#>`_
+
+    """
+
+    def __init__(self, root: str, **kwargs):
+        super().__init__(
+            root=root,
+            remote_urls="https://archive.ics.uci.edu/ml/machine-learning-databases/00224/Dataset.zip",
+            **kwargs
+        )
+
+    def _standardize(self):
+        # the file retrieved from the UCSD repository needs to be unzipped.
+        with ZipFile(os.path.join(self.root, "Dataset.zip"), "r") as zf:
```

Comment:
This assumes that the base class of split by the last backslash, which it does. But, we shouldn't assume such internal details so it's better to explicitly include the local filename for the downloaded file in the instance variables as it's downloaded. You can see e.g. how torchvision does that. Because potentially, there may not even be a single file downloaded but many so splitting by the last may be incorrect. 

But that kind of goes along with my issue that we can't really assume much in how the data comes so the base class should mostly be bare. So I'd even potentially argue that the base `Data` class should not do anything and leave all its methods as pass, and potentially even fewer methods with most of the stuff there instead only being local to Drift!

Reply:
> there may not even be a single file downloaded

Yes, but this implementation is in `DriftDataset`, not in the base class. `DriftDataset` comes from a specific (hard coded) URL, as a single ZIP archive with a specific name.